mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -90,22 +93,34 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
     TaskAttemptContext context = TezUtil.enrichContextWithAttemptWrapper(originalContext);
 
     TaskAttemptID attemptID = context.getTaskAttemptID();
-    String fileForCommitLocation = generateFileForCommitLocation(context.getJobConf(),
-        attemptID.getJobID(), attemptID.getTaskID().getId());
-    HiveIcebergRecordWriter writer = HiveIcebergRecordWriter.getWriter(attemptID);
+    JobConf jobConf = context.getJobConf();
+    Map<String, HiveIcebergRecordWriter> writers = HiveIcebergRecordWriter.getWriters(attemptID);
+    Map<String, String> outputs = SerializationUtil.deserializeFromBase64(jobConf.get(InputFormatConfig.OUTPUT_TABLES));
 
-    DataFile[] closedFiles;
-    if (writer != null) {
-      closedFiles = writer.dataFiles();
-    } else {
-      closedFiles = new DataFile[0];
+    ExecutorService tableExecutor = tableExecutor(jobConf, outputs.size());
+    try {
+      // Generates commit files for the target tables in parallel
+      Tasks.foreach(outputs.entrySet())
+          .stopOnFailure()
```

Comment:
do we want to use `retry()` here and in commitJob?

Reply:
I do not see the enough added value for the complexity:
- Here: Temporary outage of the FS could be mitigated - but we should handle if the file already exists
- `commitJob`: Reading files are already done with retries - but we should handle if we already committed the changes to the table

Do I miss something?

Reply:
For `commitJob`, you're right, most FS problems would already be retried when reading the forCommit files. For `commitTask`, I thought failing to write the forCommit file due to some intermittent FS issue could be handled and safely retried. Does it add too much to the complexity? I though at least for commitTask it wouldn't, we would just use the retry call in the fluent API.

Reply:
I was assuming that CreateFileFor commit should handle retries too, like not throw if the file already exists. Will check this soon...

Reply:
Ok. So since we use `fileForCommit.createOrOverwrite()` in `createFileForCommit` we can retry without problems. So added retry here.